extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -709,14 +690,35 @@ export function resetSraStateForTesting() {
 
 /**
  * @param {!Element} element
- * @return {string} networkId from data-ad-slot attribute.
+ * @return {string} networkId from data-slot or json attribute.
  * @visibileForTesting
  */
 export function getNetworkId(element) {
-  const networkId = /^(?:\/)?(\d+)/.exec(
-      dev().assertString(element.getAttribute('data-slot')));
-  // TODO: guarantee data-ad-slot format as part of isValidElement?
-  return networkId ? networkId[1] : '';
+  // TODO(@taymonbeal, #10524): unify this with methods in AmpA4A
+  let slot;
+  const jsonAttribute = element.getAttribute('json');
+  if (jsonAttribute) {
+    const jsonConfig = tryParseJson(jsonAttribute, err => {
+      user().warn(TAG, 'JSON invalid syntax', err && err.message);
+    });
+    if (jsonConfig && typeof jsonConfig['slot'] == 'string') {
+      slot = jsonConfig['slot'];
+    } else {
+      user().warn(TAG, 'slot not found in JSON', jsonConfig);
```

Comment:
Why?  shouldn't we fall back to the attribute value?  Something like:

const networkId = /^(?:\/)?(\d+)/.exec( 
    tryParseJson(element.getAttribute('json'), {})['slot'] || 
    element.getAttribute('data-slot'));